src/envoy/http/mixer/check_data.cc

```diff
@@ -200,6 +200,24 @@ bool CheckData::GetAuthenticationResult(istio::authn::Result* result) const {
   return Utils::Authentication::FetchResultFromHeader(headers_, result);
 }
 
+std::string CheckData::GetUrlPath() const {
+  if (headers_.Path()) {
+    const HeaderString& path = headers_.Path()->value();
+    const char* query_start = Utility::findQueryStringStart(path);
+    if (query_start != nullptr) {
+      return std::string(path.c_str(), query_start - path.c_str());
+    }
```

Comment:
need to handle query_start == nullptr case

Reply:
Done.